### ISSUE TITLE
Nightly test failure installing IPA with CA service (rawhide)

### ISSUE THREAD

**Issue by flo-renaud** [dogtagpki/pki#3927]
### Issue
The IPA nightly tests running on rawhide are failing in the server installation, in the step configuring the CA server. See PR #[1556](https://github.com/freeipa-pr-ci2/freeipa/pull/1556) with for instance the test `test_installation_TestInstallMaster`: [Details](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/f70259ac-9c75-11ec-8cbd-fa163ed4149f/), [report](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/f70259ac-9c75-11ec-8cbd-fa163ed4149f/report.html).

```
RUN ['ipa-server-install', '-n', 'ipa.test', '-r', 'IPA.TEST', '-p', 'Secret.123', '-a', 'Secret.123', '--domain-level=1', '--dirsrv-config-file', '/ipatests/ipatests_dse.ldif', '-U']
[...]
Configuring certificate server (pki-tomcatd). Estimated time: 3 minutes
  [1/29]: configuring certificate server instance
Failed to configure CA instance
See the installation logs and the following files/directories for more information:
  /var/log/pki/pki-tomcat
  [error] RuntimeError: CA configuration failed.
CA configuration failed.
The ipa-server-install command failed. See /var/log/ipaserver-install.log for more information
Exit code: 1
```

The log file [pki-ca-spawn](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/f70259ac-9c75-11ec-8cbd-fa163ed4149f/test_integration-test_installation.py-TestInstallMaster-test_install_master/master.ipa.test/var/log/pki/pki-ca-spawn.20220305114301.log.gz) shows that the pkispawn command fails waiting for PKI server to start:
```
2022-03-05 11:44:23 DEBUG: Command: systemctl start pki-tomcatd@pki-tomcat.service
2022-03-05 11:44:27 INFO: Waiting for PKI server to start
...
2022-03-05 11:46:26 INFO: Waiting for PKI server to start (119s)
```

And the journal shows the following:
```
Mar 05 11:44:27 master.ipa.test audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=pki-tomcatd@pki-tomcat comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Mar 05 11:44:27 master.ipa.test server[18886]: Java virtual machine used: /usr/lib/jvm/jre-17-openjdk/bin/java
Mar 05 11:44:27 master.ipa.test server[18886]: classpath used: /usr/share/tomcat/bin/bootstrap.jar:/usr/share/tomcat/bin/tomcat-juli.jar:
Mar 05 11:44:27 master.ipa.test server[18886]: main class used: org.apache.catalina.startup.Bootstrap
Mar 05 11:44:27 master.ipa.test server[18886]: flags used: -Dcom.redhat.fips=false
Mar 05 11:44:27 master.ipa.test server[18886]: options used: -Dcatalina.base=/var/lib/pki/pki-tomcat -Dcatalina.home=/usr/share/tomcat -Djava.endorsed.dirs= -Djava.io.tmpdir=/var/lib/pki/pki-tomcat/temp -Djava.util.logging.config.file=/var/lib/pki/pki-tomcat/conf/logging.properties -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager     -Djava.security.manager     -Djava.security.policy==/var/lib/pki/pki-tomcat/conf/catalina.policy
Mar 05 11:44:27 master.ipa.test server[18886]: arguments used: start
Mar 05 11:44:27 master.ipa.test server[18886]: WARNING: A command line option has enabled the Security Manager
Mar 05 11:44:27 master.ipa.test server[18886]: WARNING: The Security Manager is deprecated and will be removed in a future release
Mar 05 11:44:32 master.ipa.test server[18886]: WARNING: Some of the specified [protocols] are not supported by the SSL engine and have been skipped: [[TLSv1, TLSv1.1]]
Mar 05 11:44:46 master.ipa.test server[18886]: WARNING: The SHA1 algorithm used in com.netscape.ca.serviceCheckChallenge::<init>:1631 is deprecated. Use a more secure algorithm.
Mar 05 11:44:46 master.ipa.test server[18886]: WARNING: The SHA1 algorithm used in com.netscape.cmscore.authentication.ChallengePhraseAuthentication::init:116 is deprecated. Use a more secure algorithm.
Mar 05 11:44:47 master.ipa.test server[18886]: SEVERE: The required Server component failed to start so Tomcat is unable to start.
Mar 05 11:44:47 master.ipa.test server[18886]: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina]]
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.util.LifecycleBase.handleSubClassException(LifecycleBase.java:440)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:198)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.core.StandardService.startInternal(StandardService.java:432)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:927)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.startup.Catalina.start(Catalina.java:772)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:476)
Mar 05 11:44:47 master.ipa.test server[18886]: Caused by: java.lang.ExceptionInInitializerError
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.tomcat.util.threads.TaskThreadFactory.newThread(TaskThreadFactory.java:63)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.<init>(ThreadPoolExecutor.java:630)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:920)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.util.concurrent.ThreadPoolExecutor.ensurePrestart(ThreadPoolExecutor.java:1593)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:346)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.scheduleWithFixedDelay(ScheduledThreadPoolExecutor.java:680)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.tomcat.util.threads.ScheduledThreadPoolExecutor.scheduleWithFixedDelay(ScheduledThreadPoolExecutor.java:138)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:951)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:263)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
Mar 05 11:44:47 master.ipa.test server[18886]:         ... 11 more
Mar 05 11:44:47 master.ipa.test server[18886]: Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private java.security.AccessControlContext java.lang.Thread.inheritedAccessControlContext accessible: module java.base does not "opens java.lang" to unnamed module @6615435c
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
Mar 05 11:44:47 master.ipa.test server[18886]:         at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
Mar 05 11:44:47 master.ipa.test server[18886]:         at org.apache.tomcat.util.security.PrivilegedSetAccessControlContext.<clinit>(PrivilegedSetAccessControlContext.java:41)
Mar 05 11:44:47 master.ipa.test server[18886]:         ... 21 more
Mar 05 11:44:47 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.io=null" to the JVM command line arguments to enable ObjectStream cache memory leak protection. Alternatively, you can suppress this warning by disabling ObjectStream class cache memory leak protection.
Mar 05 11:44:47 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.lang=null" to the JVM command line arguments to enable ThreadLocal memory leak detection. Alternatively, you can suppress this warning by disabling ThreadLocal memory leak detection.
Mar 05 11:44:47 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.rmi/sun.rmi.transport=null" to the JVM command line arguments to enable RMI Target memory leak detection. Alternatively, you can suppress this warning by disabling RMI Target memory leak detection.
Mar 05 11:44:47 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.io=null" to the JVM command line arguments to enable ObjectStream cache memory leak protection. Alternatively, you can suppress this warning by disabling ObjectStream class cache memory leak protection.
Mar 05 11:44:47 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.lang=null" to the JVM command line arguments to enable ThreadLocal memory leak detection. Alternatively, you can suppress this warning by disabling ThreadLocal memory leak detection.
Mar 05 11:44:47 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.rmi/sun.rmi.transport=null" to the JVM command line arguments to enable RMI Target memory leak detection. Alternatively, you can suppress this warning by disabling RMI Target memory leak detection.
Mar 05 11:44:48 master.ipa.test server[18886]: WARNING: The web application [ca] appears to have started a thread named [AsyncLoader watchdog] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
Mar 05 11:44:48 master.ipa.test server[18886]:  java.base@17.0.2/java.lang.Object.wait(Native Method)
Mar 05 11:44:48 master.ipa.test server[18886]:  java.base@17.0.2/java.lang.Object.wait(Object.java:338)
Mar 05 11:44:48 master.ipa.test server[18886]:  java.base@17.0.2/java.util.TimerThread.mainLoop(Timer.java:537)
Mar 05 11:44:48 master.ipa.test server[18886]:  java.base@17.0.2/java.util.TimerThread.run(Timer.java:516)
Mar 05 11:44:49 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.io=null" to the JVM command line arguments to enable ObjectStream cache memory leak protection. Alternatively, you can suppress this warning by disabling ObjectStream class cache memory leak protection.
Mar 05 11:44:49 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.lang=null" to the JVM command line arguments to enable ThreadLocal memory leak detection. Alternatively, you can suppress this warning by disabling ThreadLocal memory leak detection.
Mar 05 11:44:49 master.ipa.test server[18886]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.rmi/sun.rmi.transport=null" to the JVM command line arguments to enable RMI Target memory leak detection. Alternatively, you can suppress this warning by disabling RMI Target memory leak detection.
```

Installed packages (full list available [here](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/f70259ac-9c75-11ec-8cbd-fa163ed4149f/installed_packages/installed_packages_master.log.gz)):

389-ds-base-2.1.0-1.fc36.x86_64
dogtag-pki-server-11.1.0-1.fc37.noarch
dogtag-pki-symkey-11.0.0-1.fc36.x86_64
tomcat-9.0.59-1.fc37.noarch
tomcatjss-8.1.0-1.fc37.noarch
java-11-openjdk-headless-11.0.14.1.1-5.fc37.x86_64
java-17-openjdk-headless-17.0.2.0.8-7.fc37.x86_64

Companion issue on IPA side: https://pagure.io/freeipa/issue/9122

**Comment by flo-renaud** [comment]
The issue seems related to tomcat update to  tomcat-9.0.59-1.fc37.noarch. Downgrading tomcat to 9.0.56 solves the issue.

**Comment by csutherl** [comment]
It appears that the combination of the fix for [ASF BZ 65714](https://bz.apache.org/bugzilla/show_bug.cgi?id=65714) and the use of Java 17 is causing the issue. A fix for asfbz#65714 was introduced in 9.0.57 (which you can see [here](https://github.com/apache/tomcat/commit/9eb49f93) and [here](https://github.com/apache/tomcat/commit/115334b1b5c)) that ensured newly created threads inherited the appropriate AccessControlContext when running under the SecurityManager. JDK versions prior to Java 17 were much more permissive in this area, but Java 17 is strict and disallows access unless otherwise configured. This issue also doesn't affect the vanilla Apache Tomcat distribution because [bin/catalina.sh already contains the necessary --add-opens calls](https://github.com/apache/tomcat/blob/9.0.x/bin/catalina.sh#L326) which allows the access to occur. We should add the same parameters to the tomcat.conf so that the reflection is allowed and Tomcat can function normally; I did test this and it seems to resolve the issue and successfully completes the ipa-server-install call.

**Comment by AdamWill** [comment]
Note, openQA found that FreeIPA often fails to start when you upgrade an *already-deployed* server to tomcat 9.0.59 also. The following log messages are in the journal. Do we need to do anything more to handle upgrade cases as well as new-deployment cases?

    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org systemd[1]: Started pki-tomcatd@pki-tomcat.service - PKI Tomcat Server pki-tomcat.
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=pki-tomcatd@pki-tomcat comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org server[3051]: Java virtual machine used: /usr/lib/jvm/jre-17-openjdk/bin/java
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org server[3051]: classpath used: /usr/share/tomcat/bin/bootstrap.jar:/usr/share/tomcat/bin/tomcat-juli.jar:
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org server[3051]: main class used: org.apache.catalina.startup.Bootstrap
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org server[3051]: flags used: -Dcom.redhat.fips=false
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org server[3051]: options used: -Dcatalina.base=/var/lib/pki/pki-tomcat -Dcatalina.home=/usr/share/tomcat -Djava.endorsed.dirs= -Djava.io.tmpdir=/var/lib/pki/pki-tomcat/temp -Djava.util.logging.config.file=/var/lib/pki/pki-tomcat/conf/logging.properties -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager     -Djava.security.manager     -Djava.security.policy==/var/lib/pki/pki-tomcat/conf/catalina.policy
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org server[3051]: arguments used: start
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: A command line option has enabled the Security Manager
    Mar 07 02:36:40 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: The Security Manager is deprecated and will be removed in a future release
    Mar 07 02:36:41 ipa001.test.openqa.fedoraproject.org pcscd[2161]: 02095788 auth.c:137:IsClientAuthorized() Process 3051 (user: 17) is NOT authorized for action: access_pcsc
    Mar 07 02:36:41 ipa001.test.openqa.fedoraproject.org pcscd[2161]: 00000297 winscard_svc.c:335:ContextThread() Rejected unauthorized PC/SC client
    Mar 07 02:36:41 ipa001.test.openqa.fedoraproject.org pcscd[2161]: 00040568 auth.c:137:IsClientAuthorized() Process 3051 (user: 17) is NOT authorized for action: access_pcsc
    Mar 07 02:36:41 ipa001.test.openqa.fedoraproject.org pcscd[2161]: 00000190 winscard_svc.c:335:ContextThread() Rejected unauthorized PC/SC client
    Mar 07 02:36:41 ipa001.test.openqa.fedoraproject.org pcscd[2161]: 00040523 auth.c:137:IsClientAuthorized() Process 3051 (user: 17) is NOT authorized for action: access_pcsc
    Mar 07 02:36:41 ipa001.test.openqa.fedoraproject.org pcscd[2161]: 00000651 winscard_svc.c:335:ContextThread() Rejected unauthorized PC/SC client
    Mar 07 02:36:41 ipa001.test.openqa.fedoraproject.org pcscd[2161]: 00041781 auth.c:137:IsClientAuthorized() Process 3051 (user: 17) is NOT authorized for action: access_pcsc
    Mar 07 02:36:41 ipa001.test.openqa.fedoraproject.org pcscd[2161]: 00000647 winscard_svc.c:335:ContextThread() Rejected unauthorized PC/SC client
    Mar 07 02:36:42 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: Some of the specified [protocols] are not supported by the SSL engine and have been skipped: [[TLSv1, TLSv1.1]]
    Mar 07 02:36:46 ipa001.test.openqa.fedoraproject.org ns-slapd[2405]: [07/Mar/2022:05:36:46.711707018 -0500] - ERR - ipa-topology-plugin - ipa_topo_util_get_replica_conf: server configuration missing
    Mar 07 02:36:46 ipa001.test.openqa.fedoraproject.org ns-slapd[2405]: [07/Mar/2022:05:36:46.716634612 -0500] - ERR - ipa-topology-plugin - ipa_topo_util_get_replica_conf: cannot create replica
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: The SHA1 algorithm used in com.netscape.ca.serviceCheckChallenge::<init>:1631 is deprecated. Use a more secure algorithm.
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: The SHA1 algorithm used in com.netscape.cmscore.authentication.ChallengePhraseAuthentication::init:116 is deprecated. Use a more secure algorithm.
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: SEVERE: The required Server component failed to start so Tomcat is unable to start.
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina]]
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.util.LifecycleBase.handleSubClassException(LifecycleBase.java:440)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:198)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.core.StandardService.startInternal(StandardService.java:432)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:927)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.startup.Catalina.start(Catalina.java:772)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:476)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: Caused by: java.lang.ExceptionInInitializerError
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.tomcat.util.threads.TaskThreadFactory.newThread(TaskThreadFactory.java:63)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.<init>(ThreadPoolExecutor.java:630)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:920)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.util.concurrent.ThreadPoolExecutor.ensurePrestart(ThreadPoolExecutor.java:1593)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:346)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.scheduleWithFixedDelay(ScheduledThreadPoolExecutor.java:680)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.tomcat.util.threads.ScheduledThreadPoolExecutor.scheduleWithFixedDelay(ScheduledThreadPoolExecutor.java:138)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:951)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:263)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         ... 11 more
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private java.security.AccessControlContext java.lang.Thread.inheritedAccessControlContext accessible: module java.base does not "opens java.lang" to unnamed module @35851384
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         at org.apache.tomcat.util.security.PrivilegedSetAccessControlContext.<clinit>(PrivilegedSetAccessControlContext.java:41)
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]:         ... 21 more
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.io=null" to the JVM command line arguments to enable ObjectStream cache memory leak protection. Alternatively, you can suppress this warning by disabling ObjectStream class cache memory leak protection.
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.lang=null" to the JVM command line arguments to enable ThreadLocal memory leak detection. Alternatively, you can suppress this warning by disabling ThreadLocal memory leak detection.
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.rmi/sun.rmi.transport=null" to the JVM command line arguments to enable RMI Target memory leak detection. Alternatively, you can suppress this warning by disabling RMI Target memory leak detection.
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.io=null" to the JVM command line arguments to enable ObjectStream cache memory leak protection. Alternatively, you can suppress this warning by disabling ObjectStream class cache memory leak protection.
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.lang=null" to the JVM command line arguments to enable ThreadLocal memory leak detection. Alternatively, you can suppress this warning by disabling ThreadLocal memory leak detection.
    Mar 07 02:36:47 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.rmi/sun.rmi.transport=null" to the JVM command line arguments to enable RMI Target memory leak detection. Alternatively, you can suppress this warning by disabling RMI Target memory leak detection.
    Mar 07 02:36:48 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: The web application [ca] appears to have started a thread named [AsyncLoader watchdog] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
    Mar 07 02:36:48 ipa001.test.openqa.fedoraproject.org server[3051]:  java.base@17.0.2/java.lang.Object.wait(Native Method)
    Mar 07 02:36:48 ipa001.test.openqa.fedoraproject.org server[3051]:  java.base@17.0.2/java.lang.Object.wait(Object.java:338)
    Mar 07 02:36:48 ipa001.test.openqa.fedoraproject.org server[3051]:  java.base@17.0.2/java.util.TimerThread.mainLoop(Timer.java:537)
    Mar 07 02:36:48 ipa001.test.openqa.fedoraproject.org server[3051]:  java.base@17.0.2/java.util.TimerThread.run(Timer.java:516)
    Mar 07 02:36:48 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.io=null" to the JVM command line arguments to enable ObjectStream cache memory leak protection. Alternatively, you can suppress this warning by disabling ObjectStream class cache memory leak protection.
    Mar 07 02:36:48 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.base/java.lang=null" to the JVM command line arguments to enable ThreadLocal memory leak detection. Alternatively, you can suppress this warning by disabling ThreadLocal memory leak detection.
    Mar 07 02:36:48 ipa001.test.openqa.fedoraproject.org server[3051]: WARNING: When running on Java 9 or later you need to add "--add-opens=java.rmi/sun.rmi.transport=null" to the JVM command line arguments to enable RMI Target memory leak detection. Alternatively, you can suppress this warning by disabling RMI Target memory leak detection.

**Comment by csutherl** [comment]
@AdamWill the solution was to add a new conf file (see https://src.fedoraproject.org/rpms/tomcat/c/9b5ef6dbd7adb3dca35411e43d1f8ba9517b8010?branch=rawhide) which will resolve upgrade or new install issues. The conf file is read by the service wrapper, so it doesn't matter if you override tomcat.conf with pki-tomcat.conf file, it still gets loaded.

**Comment by flo-renaud** [comment]
The issue is tracked on tomcat side at https://bugzilla.redhat.com/show_bug.cgi?id=2061424 and has been solved with [tomcat-9.0.59-2.fc37](https://bodhi.fedoraproject.org/updates/FEDORA-2022-bfc7b60b73).
Successful run in PR#[1571](https://github.com/freeipa-pr-ci2/freeipa/pull/1571), test fedora-rawhide/test_installation_TestInstallWithoutNamed with the following [logs](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/d71bef92-a1fa-11ec-82f2-fa163e2b0c01/) and [report](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/d71bef92-a1fa-11ec-82f2-fa163e2b0c01/report.html).

This issue can be closed.
